### PR TITLE
AB#657: title page

### PR DESCRIPTION
**Pull request by damon-m-deng**
[AB#658](https://dev.azure.com/2208SDE07/0fe9ffb8-4ebd-425e-9462-cb672e14673f/_workitems/edit/658): The player can press enter (or click the button) to enter the game

-- Improved workflow
-- Initial page->press enter->new page with intro story and start button->main text page (stats)

### CODE REVIEW

File: src/com/awakening/gui/app/GameHomePage.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public static JTextArea getHomePageTextArea() {
+        return homePageTextArea;
+    }
+
+    public static JButton getGameStartButton() {
+        return gameStartButton;
+    }
+}
```
Comment (damon-m-deng) [inline src/com/awakening/gui/app/GameHomePage.java:63]: I made this page completely separate from the title page which contains the picture and the "press enter" button. Therefore, I was able to clear out the previous panels.

File: src/com/awakening/gui/app/GameStart.java [hunk 2]
```diff
@@ -25,55 +22,44 @@ public GameStart(){
...
-        label_title = new JLabel(icon);
+        splash_label = new JLabel(icon);
 
-        // start button
-        startButtonPanel = new JPanel();
-        startButtonPanel.setBounds(250, 630, 400, 200);
-        startButtonPanel.setBackground(Color.black);
+        // press enter to start button
+        pressEnterToStartButton_panel = new JPanel();
+        pressEnterToStartButton_panel.setBounds(250, 630, 400, 200);
```
Comment (damon-m-deng) [inline src/com/awakening/gui/app/GameStart.java:35]: All of the positioning and bounds are static at the moment, therefore:
- elements are not auto-adjusted by screen resolution
- difficult to implement... A lot of trial&errors to find the right position

File: src/com/awakening/gui/app/GameStart.java [hunk 3]
```diff
@@ -25,55 +22,44 @@ public GameStart(){
...
-        start_button.setBackground(Color.black);
-        start_button.setForeground(Color.lightGray);
-        start_button.setFont(Awakening_Font.getNormalFont());
-        start_button.addActionListener(e -> ProgressScreen.startProgressScreen(this));
+        pressEnterToContinueButton = new JButton("PRESS ENTER TO START");
+        pressEnterToContinueButton.setBackground(Color.black);
+        pressEnterToContinueButton.setForeground(Color.lightGray);
+        pressEnterToContinueButton.setFont(Awakening_Font.getNormalFont());
+        pressEnterToContinueButton.setFocusPainted(false);
+        pressEnterToContinueButton.addActionListener(e -> GameHomePage.createHomePage());
```
Comment (damon-m-deng) [inline src/com/awakening/gui/app/GameStart.java:43]: got rid of the argument of the ActionListener, used static method/fields instead

**Review by Legatus727 — APPROVED** [review]
Awesome job Damon!

File: src/com/awakening/gui/app/GameStartButtonHandler.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.awt.event.ActionListener;
+
+public class GameStartButtonHandler implements ActionListener {
+
+    public static void showTextPage(){
+        GameHomePage.getHomePageTextArea().setText("AHHHHH!!!!!!");
+        GameHomePage.getHomePageTextArea().setForeground(Color.RED);
+        GameHomePage.getGameStartButton().setText("Continue");
+    }
+
```
Comment (damon-m-deng) [inline src/com/awakening/gui/app/GameStartButtonHandler.java:14]: After clicking the *start* button, the panels remain the same, but the content of the textbox has been overwritten.

**Review by javidur — APPROVED** [review]

**Pull request merged**